app/lib/shared/utils.dart

```diff
@@ -305,3 +306,53 @@ extension ByteFolderExt on Stream<List<int>> {
     return buffer.toBytes();
   }
 }
+
+/// Tracks the maximum latency by observing each latency value and keeping the maximum.
+/// The tracked maximum value decays, halving its value in every minute.
+class DecayingMaxLatencyTracker {
+  final Duration _halfLifePeriod;
+
+  int _value = 0;
+  DateTime _lastUpdated = clock.now();
+
+  DecayingMaxLatencyTracker({
+    Duration? halfLifePeriod,
+  }) : _halfLifePeriod = halfLifePeriod ?? Duration(minutes: 1);
+
+  void _decay({
+    required DateTime? now,
+    Duration? updateDelay,
+  }) {
+    now ??= clock.now();
```

Comment:
I mean a non-null required argument. There should be no reason for this helper method to access `clock.now`